packages/core/src/tracing/instrumentation/loggers/console.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. and contributors 2021
+ */
+
+/* eslint-disable max-len */
+
+'use strict';
+
+const shimmer = require('shimmer');
+const tracingUtil = require('../../tracingUtil');
+const constants = require('../../constants');
+const cls = require('../../cls');
+
+let isActive = false;
+
+exports.init = function init() {
+  shimmer.wrap(console, 'warn', shimLog({ markAsError: false, level: 'warn' }));
+  shimmer.wrap(console, 'error', shimLog({ markAsError: true, level: 'error' }));
+};
+
+// ATTENTION: Do not use console.warn or console.error in this file, otherwise it could run into endless loop
+function shimLog(options) {
+  return originalLog =>
+    function () {
+      if (arguments.length === 0) {
+        // * arguments.length === 0 -> This is a console.warn() type of call (without arguments), this will not log
+        // anything but simply return whether the log level in question is enabled for this logger.
+        return originalLog.apply(this, arguments);
+      }
+
+      if (isActive && cls.isTracing()) {
+        const parentSpan = cls.getCurrentSpan();
+
+        if (parentSpan && !constants.isExitSpan(parentSpan)) {
+          const originalArgs = new Array(arguments.length);
+
+          for (let i = 0; i < arguments.length; i++) {
+            originalArgs[i] = arguments[i];
+          }
+
+          instrumentedLog(this, originalLog, originalArgs, options);
+        } else {
+          return originalLog.apply(this, arguments);
+        }
+      } else {
+        return originalLog.apply(this, arguments);
+      }
+    };
+}
+
+function instrumentedLog(ctx, originalLog, originalArgs, options) {
+  return cls.ns.runAndReturn(() => {
+    const span = cls.startSpan('log.console', constants.EXIT);
+
+    span.stack = tracingUtil.getStackTrace(instrumentedLog);
+    const fields = originalArgs[0];
```

Comment:
*nitpick*: I think the term `fields` in the bunyan instrumentation is somewhat specific to bunyan: https://github.com/trentm/node-bunyan#log-record-fields. We also do not use that term in any other logging instrumentation. I find the use of that term here a bit non-intuitive. Both [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Console/warn) and the [Node.js docs](https://nodejs.org/api/console.html#console_console_warn_data_args) just mention an arbitrary number of parameters that can be passed to console.warn/error. Can we find a better name for this?

Reply:
Yeah good idea. Agree!

Maybe I'll just go with

```
const firstArg = originalArgs[0];
const secondArg = originalArgs[1];
```

or similar.

Reply:
Sounds good.